docs/website/contents/for-developers/EraTransitionGovernance.md

```diff
@@ -0,0 +1,140 @@
+  Byron on-chain governance as described in this document will replace it, although it will require two epochs, one each to reach protocol versions 1 and 2.
+
+- `cardano-db-sync`.
+  The DB Sync Team creates a "mock" chain to use when testing their tool.
+  [This test](https://github.com/IntersectMBO/cardano-db-sync/blob/0f1d93f9b868caaf14c6cd0e77991e63c07c067f/cardano-chain-gen/test/Test/Cardano/Db/Mock/Unit/Conway/Governance.hs#L451) raises the protocol version within Conway.
+  And [this test](https://github.com/IntersectMBO/cardano-db-sync/blob/0f1d93f9b868caaf14c6cd0e77991e63c07c067f/cardano-chain-gen/test/Test/Cardano/Db/Mock/Unit/Conway/Other.hs#L447) raises the protocol version within Babbage.
+  The linked definitions use more combinators defined elsewhere in that codebase, but it's only a few indirections until the definitions start using the `cardano-ledger` interface directly.
+  Do note that these transactions have no witnesses, which can be sufficient for the testing of a tool that trusts its upstream node.
+
+- `ouroboros-consensus`.
+  The Consensus Team wrote tests shortly before the end of the Byron era that tested the mechanism that would transition to Shelley.
+  In particular, [the `mkProtocolByronAndHardForkTxs` function](https://github.com/IntersectMBO/ouroboros-consensus/blob/a9a5f3aaf3ddd45b3dd58a132d65b657bbf285e5/ouroboros-consensus-cardano/src/unstable-byron-testlib/Test/ThreadNet/Infra/Byron/TrackUpdates.hs#L318-L363) and the `mkHardForkProposal` function defined just below it use the `cardano-ledger` interface for Byron in order to create the transactions for the proposal and votes --- each with the necessary witnesses/signatures --- in order to increment the protocol version via Byron on-chain governance.
+  Those functions are used within the Consensus Layer's "ThreadNet" tests, which submit the transactions listed in the `tniCrucialTxs` field at every possible opportunity.
+  That degenerate persistence and eagerness reliably ensures that the chain increments the protocol version as soon as possible.
+
+  A similar ThreadNet test transitions from Shelley to Allegra; its `tniCrucialTxs` field is created by [this function](https://github.com/IntersectMBO/ouroboros-consensus/blob/a9a5f3aaf3ddd45b3dd58a132d65b657bbf285e5/ouroboros-consensus-cardano/src/unstable-shelley-testlib/Test/ThreadNet/Infra/Shelley.hs#L431), which leverages the fact that a single Shelley transaction can demonstrate the necesary quorum for the protocol version increment.
+
+The key features of these three suites with regard to this document are summarized in the following table.
+
+| Repository | Byron | Shelley, ..., and/or Babbage | Conway |
+| - | - | - | - |
+| `cardano-db-sync` | No | Haskell, unsigned | Haskell, unsigned |
+| `ouroboros-consensus` | Haskell, signed | Haskell, signed | No |
```

Comment:
FTR: We have #1065 tracking work to have a Conway test in ouroboros-consensus. (Main thing there is that there currently is no era "after" Conway, so it is a bit annoying (but not impossible) to actually test that ATM.)

Reply:
I added a parenthetical reference.